src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts

```diff
@@ -179,6 +262,220 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
+                        proxyICorDebugPort,
+                        proxyBrowserPort,
+                    ];
+                }
+            }
+
+            for await (const error of spawnedProxy.stderr) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(`ERROR: ${error}`);
+            }
+        } catch (error: any) {
+            if (spawnedProxy.pid) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(
+                    `Error occured while spawning debug proxy. Terminating debug proxy server.`
+                );
+                process.kill(spawnedProxy.pid);
+            }
+            throw error;
+        }
+        return ['', -1, -1];
+    }
+
+    private static async isNet9OrNewer(): Promise<boolean> {
+        let isNet9OrNewer = false;
+        const configurationsObject = {
+            $properties: ['BuildProperties'],
+            BuildProperties: {
+                $properties: ['Id', 'Name', 'Value', 'StorageType'],
+                $where: {
+                    '&&': [
+                        { '==': [{ '.': 'Name' }, 'TargetFramework'] },
+                        { '==': [{ '.': 'StorageType' }, 'ProjectFile'] },
+                    ],
+                },
+            },
+        };
+        const query = {
+            context: 'Projects',
+            query: {
+                $properties: ['ActiveConfigurations'],
+                ActiveConfigurations: configurationsObject,
+                $filter: { ProjectsByCapabilities: ['WebAssembly'] },
+            },
+        };
+
+        const queryString = JSON.stringify(query);
+        const proxy = await getCSharpDevKit()?.exports.serviceBroker.getProxy<IQueryExecutionService>(
+            Descriptors.projectQueryExecutionService
+        );
+        try {
+            if (proxy) {
+                const result = await proxy.ExecuteQueryAsync(queryString);
+                const queryResult = JSON.parse(result) as QueryResult<Project>;
+                if (queryResult && queryResult.Items) {
+                    isNet9OrNewer = true;
+                    queryResult.Items.forEach((project) => {
+                        project.ActiveConfigurations?.forEach((activeConfig) => {
+                            if (
+                                activeConfig.BuildProperties[0].Value == 'net5.0' ||
+                                activeConfig.BuildProperties[0].Value == 'net6.0' ||
+                                activeConfig.BuildProperties[0].Value == 'net7.0' ||
+                                activeConfig.BuildProperties[0].Value == 'net8.0'
+                            ) {
+                                isNet9OrNewer = false;
+                            }
+                        });
+                    });
+                    return isNet9OrNewer;
+                }
+            }
```

Comment:
Do you need some sort of `else throw devkit-not-ready` here in case you can't get the proxy yet, but dev kit is installed?

Reply:
If the newer devkit is installed and we try to call tryToUseVSDbgForMono this function will not exist in C# extension yet and this is addressed on devkit side. if the function exists the vswebassembly will also exists.

Am I understanding the question wrongly?

Reply:
What I mean is -- this code will do:

```
if (proxy) {
    ....
}
// no else here, so the code will fall through and return `false`
```

I think instead the code should be doing something like:
```
if (!proxy) {
   throw new Error("Unable to obtain required service from C# Dev Kit.");
}
```

To avoid possible races if we try to start debugging and C# Dev Kit hasn't yet initalized.

Reply:
Thanks for the explanation.